hugolib/page.go

```diff
@@ -380,6 +380,17 @@ var (
 	internalSummaryDivider = []byte("HUGOMORE42")
 )
 
+// If the raw content has nothing but whitespace after the summary marker then
+// the page shouldn't be marked as truncated.  This check is simplest against
+// the raw content because different markup engines (rst and asciidoc in
+// particular) add div and p elements after the summary marker.
+func (p *Page) anyContentAfterUserDefinedSummary() bool {
+	sections := bytes.Split(p.rawContent, helpers.SummaryDivider)
```

Comment:
Performance is a good goal, and a nice feature of Hugo, and I'm happy to improve this.  This split is not actually done anywhere on the raw content, it is done on the work content once it has been modified by renderContent, but that is too late.  The same approximate amount of work is also done on the work content right before renderContent in commonConvert (after shortcode and emoji processing).  That is where the <!--more--> is replaced by the internal marker.  It should be fine to do this check there instead of on the true raw content, so I'll look into changing how that replacement is done so that the truncate flag information can be computed then.